refarch-integrations/refarch-s3-integration/refarch-s3-integration-service/src/main/java/de/muenchen/refarch/s3/integration/configuration/nfcconverter/NfcHelper.java

```diff
@@ -0,0 +1,141 @@
+package de.muenchen.refarch.s3.integration.configuration.nfcconverter;
+
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import java.text.Normalizer;
+import java.util.*;
+import java.util.stream.Collectors;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.map.CaseInsensitiveMap;
+
+/**
+ * Helper class for NFC normalisation.
+ *
+ * @see Normalizer
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@Slf4j
+public class NfcHelper {
+
+    /**
+     * Konvertieren eines String in die kanonische Unicode-Normalform (NFC)
+     *
+     * @param in Eingabe-String
+     * @return Normalisierter String.
+     * @see Normalizer#normalize(CharSequence, Normalizer.Form)
+     */
```

Comment:
Maybe translate the comments to english directly? You should check in all files again. Won't comment any further on translation stuff.